### DISCUSSION TITLE
Whow knows how to plot to multiple drives?

### DISCUSSION THREAD

**Discussion by Cminer3207** [swar/Swar-Chia-Plot-Manager#258] [General]
Somebody out there has figured this out... 
How do you get the thing to fill up a drive and then move on to a new drive without stopping, changing the YAML, and starting again.

**Comment by jhony633** [comment]
I also seen this on another user but can't figure it out

**Comment by Cminer3207** [comment]
Here is the answer:

In config.YAML File...

At bottom under job...

Single Drive format:
———————————
destination_directory: J:\Plots

Multiple Drive Format:
——————————
destination_directory: 
- J:\Plots
- K:\Plots
- L:\ Plots

Happy mining.

**Reply by paullepa** [reply]
For posterity syntax is:
(space)(space)(dash)(space)(DriveLetter)
ie.
  - J:\
 
or 
  - J:\my_temp_example

I'm new to YAML and it took me forever to figure out that YAML parser sucks and needs very well formated text.

**Reply by Cminer3207** [reply]
You’re a good person paullepa...

That (space)(space)(dash)(space) took me three attempts to figure out...

If Swar had taken the time to say that he could have saved me some headaches...

I can’t complain too much though... he did make the plotting tool for us all...

**Reply by Kali123411** [reply]
> For posterity syntax is:
> (space)(space)(dash)(space)(DriveLetter)
> ie.
> 
> * J:\
> 
> or
> 
> * J:\my_temp_example
> 
> I'm new to YAML and it took me forever to figure out that YAML parser sucks and needs very well formated text.

hmmm this is not working for me..

**Reply by paullepa** [reply]
try using yaml validator - it will point out for you where its having issues.
http://www.yamllint.com/

I'm new to YAML as find it very very picky

**Comment by Cminer3207** [comment]
Use the default YAML.
Second job example. 
Copy and paste the Destination Drive exactly as shown there, then change the letters to yours. 
If you have more drives, just paste it more times.

**Comment by dssamusaran** [comment]
With this multiple drives configuration, is the system handling full drives properly?

What I mean is: If a drive is full it then chooses the next one automatically.

Or do you still have to remove the full drives from the configuration?

**Reply by Cminer3207** [reply]
According to swar, the plot manager will fill the listed drives sequentially. As one fills, it switches to the next in your list. 
Any change to drive list in the YAML file will require restating the plotter after manually clearing the temporary storage drive. 
Set up all your drives and then start the plotter. 
When you want to add new destination drives to the plotter, you will need to restart. 
It’s all clearly explained in the Readme file. 
Mr Patel did a very nice job of explaining it all.

**Reply by dssamusaran** [reply]
Hello,

Thanks for your lightning fast answer.
I understood all of that quite well by reading all the readme.

Only thing I did not figure out is the behavior in case of full drive. 
When reading about the directory and temp list I got that it was doing, in case of 2 drives, plot 1 on 1, plot 2 on 2, plot 3 on 1, and so on... But not that it will be skipping to next drive in case of a full drive.

Anyway thank for the answer that clears thing up. And thanks a ton to the dev for this wonderful tool!

**Reply by dssamusaran** [reply]
Hello again.

I did some more testing with 5 drives in the list.

It is using the drive in sequence as expected. 
Plot 1 => drive 1
Plot 2 => drive 2
Plot 3 => drive 3
Plot 4 => drive 1
Plot 5 => drive 2
And so on.

But I included one drive with little space left and after a few iteration when the drive was full the manager still tries to send plots to it. The plot phases went on to the Copy phase where is was unable to copy the plot to the full drive.
The good thing is that it did not 'crash' the plot. In the log we can read this error:
Could not copy "B:\\ChiaTMP02\\plot-k32-***.plot.2.tmp" to "F:\\ChiaPlot3\\plot-k32-***.plot.2.tmp". Error There is not enough space on the disk.. Retrying in five minutes.

So I freed some space and the plot copy finished successfully without anything else to do.

It is now an issue in itself but it means to me that the destination list you use should be drives with more or less the same % of filling. Looking for this error in the log is quite mandatory to me as the plots are staying in the manager for hours without freeing the 'max_concurrent' settings you set.


Don't get me wrong I'm very very happy with the manager. I just wanted to give my test results as it may help someone else. 
It's an amazing tool that allows us all to plot very easily compared to the manual process it was without!

**Reply by Cminer3207** [reply]
Thank you for sharing this. 

So

Each Destination Drive (or group of drives with the same available space) needs its own job-order. 
Set the max_plots: parameter to the size of the available drive-space. 
maxPlots=(AvailableDriveSpace/108GB) 

This will stop creating plots just before the drive space is full and not cause a log-jam in your max_concurrent or require deleting temp storage.

**Reply by dssamusaran** [reply]
Yes this is exactly what I did. A job for each size of drives and having the max job stopping before filling up. I left around 3 plots of margin for each drives to be safe. I'll fill them up by moving files manually at the end.

**Reply by Cminer3207** [reply]
Please post a description of how you move plots between drives. 
I have heard is not as simple as just copying the .plot file to a new drive and erasing the old one. 
Thanks.

**Reply by dssamusaran** [reply]
At first I simply cut/paste from source to destination. But it happened sometimes that the moved plots were no more recognized until full restart of the chia node.

To avoid this I created folders on each drives called 'ChiaPlotInput' and 'ChiaPlotTransfered'.
Those drives are NOT checked by chia.

Here's an example of moving plots from D:\ChiaPlot to E:\ChiaPlot

-Copy the plots you want to move from D:`\ChiaPlot to E:\ChiaPlotInput (this is the long part to copy the datas)
-Once done, Move the same plots from D:\ChiaPlot to D:\ChiaPlotTransfered (same drive, instant)
-In the Chia software in the Plot tab refresh the plot directories. The plot you copied will disappear from the farm. (could be done in commande as well following what you use)
-Move the plots from E:\ChiaPlotInput to E:\ChiaPlot (same drive, instant)
-In Chia, refresh the plots and they will come back, now on the correct drive.
-Delete the plots from D:\ChiaPlotTransfered

The whole 'ChiaPlotTransfered folder is just to be safe. You could skip that and delete the plots immediately but I prefer having a failsafe in case I make a mistake. I only delete the plots from there when I see all my plots being farmed.

**Reply by Cminer3207** [reply]
Brilliant!

**Comment by Cminer3207** [comment]
It fills all the drives concurrently, instead of filling one completely and then moving on to the next.
 
D:\ 2 plot
E:\ 1 plot 
F:\ 1 plot 

etc...

**Reply by Kali123411** [reply]
> It fills all the drives concurrently, instead of filling one completely and then moving on to the next.
> 
> D:\ 2 plot
> E:\ 1 plot
> F:\ 1 plot
> 
> etc...

Yea that's why I just made separate jobs for each one of my drives.

**Reply by Cminer3207** [reply]
I prefer it this way. 
I use jobs to split the temp space and cpu power into scalable worker units based on the most efficient configuration for system hardware.
I give them a weeks worth of drive space (252 plots) connect a UPS, and forget about it as I move on to the next build. 

Changing the drive space is the only thing that bolloxes up the manager on a restart... so I give it plenty and tweak the other config parameters until it’s optimal.

**Comment by mkirshner** [comment]
Currently my setup is doing what the OP says- Swar assigns all parallel plots to one drive, fills it, then moves on.

Any advice on how to achieve the opposite- one parallel plot goes to one destination drive? I want to reduce the bottleneck from writing to the hard drive. Thanks
